CRM/Contribute/BAO/Contribution.php

```diff
@@ -3024,12 +3028,16 @@ public function _gatherMessageValues($input, &$values, $ids = []) {
             $values['priceSetID'] = $priceSet['id'];
           }
           foreach ($lineItems as &$eachItem) {
-            if (isset($this->_relatedObjects['membership'])
-              && is_array($this->_relatedObjects['membership'])
-              && array_key_exists($eachItem['entity_id'] . '_' . $eachItem['membership_type_id'], $this->_relatedObjects['membership'])) {
```

Comment:
This doesn't work because membership_type_id is not set for lineitems. The membership_type_id is never used anyway!